docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -63,248 +156,329 @@ audience. For instance:
+
+    MC->>MS: Request data
+    MS-->>MC: 401 Unauthorized with WWW-Authenticate
+    Note over MC: MCP Client retrieves and validates<br/>PRM document
+    Note over MC: MCP Client retrieves authorization<br/>endpoint from AS metadata document<br/>(Section 2.3)
+    MC-->>UA: Initiate authorization request at the authorization endpoint
+    Note over UA: User authenticates and authorizes<br/>client for scope and resource
+    UA->>MC: Return authorization code to client
+    MC->>AS: Exchange code for token
+    AS-->>MC: Access token
+    MC->>MC: Cache token
+    MC->>MS: Request data</br>(with Authorization header)
 ```
 
-#### 2.3.1 Server Metadata Discovery Headers
+#### 2.4.4 Security Considerations
+
+When implementing authorization with an existing AS, MCP servers **MUST**:
 
-MCP clients _SHOULD_ include the header `MCP-Protocol-Version: <protocol-version>` during
-Server Metadata Discovery to allow the MCP server to respond based on the MCP protocol
-version.
+1. Request the minimum number of required scopes.  
+2. Securely handle any tokens that need to be stored or otherwise managed by the MCP server.  
+3. Implement appropriate session timeout handling.  
+4. Consider security implications of token chaining, such as ensuring that the appropriate
+   tokens are exposed to the right audiences (e.g., don't pass server-issued tokens to the
+   client), validating the original scopes and issuers, and looking for unintended privilege
+   escalation by virtue of token exchange. Refer to the [Best Practices](#3-best-practices)
+   section for additional recommendations.
+5. Implement proper error handling for authorization failures.
 
-For example: `MCP-Protocol-Version: 2024-11-05`
+MCP clients **MUST**:
 
-#### 2.3.2 Authorization Base URL
+1. Securely store Access and Refresh Tokens received from the AS.  
+2. Properly handle requests to renew and replace expired or invalid tokens (using the
+   Refresh Token grant) or due to challenges (refer to
+   [RFC 9470](https://www.rfc-editor.org/rfc/rfc9470.html)).
+3. Verify that the `resource` identifier in the PRM document matches the MCP server.
 
-The authorization base URL **MUST** be determined from the MCP server URL by discarding
-any existing `path` component. For example:
+Authorization servers **MUST**:
 
-If the MCP server URL is `https://api.example.com/v1/mcp`, then:
+1. Verify that the resource identifier is registered for the audience of the access token issued.
 
-- The authorization base URL is `https://api.example.com`
-- The metadata endpoint **MUST** be at
-  `https://api.example.com/.well-known/oauth-authorization-server`
+### 2.5 Delegated Authorization or Colocated AS
 
-This ensures authorization endpoints are consistently located at the root level of the
-domain hosting the MCP server, regardless of any path components in the MCP server URL.
+In limited scenarios, MCP servers **MAY** implement their own AS, referred to as a _colocated AS_ due
```

Comment:
What does "limited" mean here?

Maybe rephrase as "MCP servers MAY implement their own AS ... though this should be limited to scenarios where there is significant authorization expertise amongst the MCP server owners."

or something of that nature.

Reply:
Great feedback. Updated.